### PR TITLE
Saving/opening

### PR DESCRIPTION
**Pull request by UAChronos**

### CODE REVIEW

File: RPGSystem/GameForm.cs [hunk 1]
```diff
@@ -55,24 +57,108 @@ private void btnChangeAmmountOfGoldAndCraftingPieces_Click(object sender, EventA
...
+            {
+                GameDataService.Instance.SaveToXml(filePath);
+            }
+            else
+            {
+                GameDataService.Instance.SaveToJson(filePath);
+            }
+        }
+
+        protected string filePath;
```
Comment (Krum78) [inline RPGSystem/GameForm.cs]: Make them private

File: RPGSystem/GameForm.cs [hunk 2]
```diff
@@ -55,24 +57,108 @@ private void btnChangeAmmountOfGoldAndCraftingPieces_Click(object sender, EventA
...
+                    }
+                    else
+                    {
+                        GameDataService.Instance.SaveToJson(fileName);
+                    }
+                }
+            }
+        }
+
+        protected void Save()
```
Comment (Krum78) [inline RPGSystem/GameForm.cs]: Make it private

File: RPGSystem/GameForm.cs [hunk 3]
```diff
@@ -55,24 +57,108 @@ private void btnChangeAmmountOfGoldAndCraftingPieces_Click(object sender, EventA
...
             lblAmmountOfGold.Text = GameDataService.Instance.GameModel.Gold.ToString();
             lblAmmountOfCraftingPieces.Text = GameDataService.Instance.GameModel.CraftingPieces.ToString();
+            if (isFullUpdate == true)
+            {
+                flowLayoutPanel1.Controls.Clear();
+                flowLayoutPanel1.Controls.AddRange(GameDataService.Instance.GameModel.Players.Select(p => new PlayerStats(p)).ToArray());
+            }
+        }
+
+        protected void SaveAs()
```
Comment (Krum78) [inline RPGSystem/GameForm.cs]: The same, wrong access modifier

File: RPGSystem/GameForm.cs [hunk 4]
```diff
@@ -5,10 +5,12 @@
 using System.ComponentModel;
 using System.Data;
 using System.Drawing;
+using System.IO;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
 using System.Windows.Forms;
+using System.Xml.Serialization;
```
Comment (Krum78) [inline RPGSystem/GameForm.cs]: Consider removing unused usings

**Review by Krum78 — CHANGES_REQUESTED** [review]

File: RPGSystem/GameForm.cs (hunk 1 above)
Comment (UAChronos) [reply RPGSystem/GameForm.cs]: Fixed

File: RPGSystem/GameForm.cs (hunk 2 above)
Comment (UAChronos) [reply RPGSystem/GameForm.cs]: Fixed

File: RPGSystem/GameForm.cs (hunk 3 above)
Comment (UAChronos) [reply RPGSystem/GameForm.cs]: Fixed

File: RPGSystem/GameForm.cs (hunk 4 above)
Comment (UAChronos) [reply RPGSystem/GameForm.cs]: Fixed

**Review by Krum78 — APPROVED** [review]

**Review by Krum78 — APPROVED** [review]

**Pull request merged**